Sources/Build/BuildDelegate.swift

```diff
@@ -368,103 +383,100 @@ public final class BuildDelegate: BuildSystemDelegate, SwiftCompilerOutputParser
 
     /// The last task text before the task list was emptied.
-    private var lastText: String?
-    
-    var latestRunningText: String? {
-        return tasks.last?.text ?? lastText
-    }
-    
-    mutating func commandStarted(_ command: SPMLLBuild.Command) {
-        addTask(identifier: command.name, text: command.description)
-        totalCount += 1
+    private(set) var latestFinishedText: String?
+
+    mutating func commandStatusChanged(_ command: SPMLLBuild.Command, kind: CommandStatusKind) {
+        switch kind {
+        case .isScanning:
+            totalCount += 1
+            break
+        case .isUpToDate:
+            totalCount -= 1
+            break
+        case .isComplete:
+            break
+        }
     }
     
-    mutating func commandFinished(_ command: SPMLLBuild.Command, result: CommandResult) {
-        removeTask(identifier: command.name)
+    mutating func commandFinished(_ command: SPMLLBuild.Command, result: CommandResult, targetName: String?) {
+        latestFinishedText = progressText(of: command, targetName: targetName)
 
         switch result {
-        case .succeeded:
+        case .succeeded, .skipped:
             finishedCount += 1
-        case .cancelled, .failed, .skipped:
+        case .cancelled, .failed:
             break
         }
     }
     
-    mutating func swiftCompilerDidOuputMessage(_ message: SwiftCompilerMessage) {
+    mutating func swiftCompilerDidOuputMessage(_ message: SwiftCompilerMessage, targetName: String) {
         switch message.kind {
         case .began(let info):
-            if let text = message.progressText {
-                addTask(identifier: info.pid.description, text: text)
+            if let text = progressText(of: message, targetName: targetName) {
+                swiftTaskProgressTexts[info.pid] = text
             }
 
             totalCount += 1
         case .finished(let info):
-            removeTask(identifier: info.pid.description)
+            if let progressText = swiftTaskProgressTexts[info.pid] {
+                latestFinishedText = progressText
+                swiftTaskProgressTexts[info.pid] = nil
+            }
+
             finishedCount += 1
-        case .signalled(let info):
-            removeTask(identifier: info.pid.description)
-        case .skipped:
+        case .signalled, .skipped:
             break
         }
     }
-    
-    private mutating func addTask(identifier: String, text: String) {
-        tasks.append(Task(identifier: identifier, text: text))
-    }
-    
-    private mutating func removeTask(identifier: String) {
-        if let index = tasks.index(where: { $0.identifier == identifier }) {
-            if tasks.count == 1 {
-                lastText = tasks[0].text
-            }
 
-            tasks.remove(at: index)
+    private func progressText(of command: SPMLLBuild.Command, targetName: String?) -> String {
+        if let firstSpaceIndex = command.description.firstIndex(of: " "),
```

Comment:
Can you document some examples of the command descriptions that are being handled here?

Reply:
I documented an example.